diff --git a/ModelDescriber/JMSModelDescriber.php b/ModelDescriber/JMSModelDescriber.php
index c3c33c5..e80d271 100644
--- a/ModelDescriber/JMSModelDescriber.php
+++ b/ModelDescriber/JMSModelDescriber.php
@@ -83,7 +83,8 @@ class JMSModelDescriber implements ModelDescriberInterface, ModelRegistryAwareIn
                 $groups = $model->getGroups()[$name];
             } elseif (!isset($groups[$name]) && !empty($this->previousGroups[spl_object_hash($model)])) {
                 // $groups = $this->previousGroups[spl_object_hash($model)]; use this for jms/serializer 2.0
-                $groups = false === $this->propertyTypeUsesGroups($item->type) ? null : [GroupsExclusionStrategy::DEFAULT_GROUP];
+                // $groups = false === $this->propertyTypeUsesGroups($item->type) ? null : [GroupsExclusionStrategy::DEFAULT_GROUP];
+                $groups = false === $this->propertyTypeUsesGroups($item->type);
             } elseif (is_array($groups)) {
                 $groups = array_filter($groups, 'is_scalar');
             }
@@ -93,7 +94,7 @@ class JMSModelDescriber implements ModelDescriberInterface, ModelRegistryAwareIn
             }
 
             // read property options from Swagger Property annotation if it exists
-            if (null !== $item->reflection) {
+            if (isset($item->reflection) && null !== $item->reflection) {
                 $property = $properties->get($annotationsReader->getPropertyName($item->reflection, $name));
                 $annotationsReader->updateProperty($item->reflection, $property, $groups);
             } else {
